Kc dev upstream 26.4

ademone opened the pull request.

Commit:
Adjust files for 26.4 version of keycloak, also add templates

Commit:
Adjust files for 26.4 version of keycloak, also add templates

NicolasLiampotis commented on roles/keycloak/templates/26.4/keycloak.conf.j2:
```diff
@@ -0,0 +1,41 @@
+
+# The proxy address forwarding mode if the server is behind a reverse proxy.
+#proxy=reencrypt
+#proxy=edge
```
We can remove these two lines since the proxy option was deprecated in Keycloak v24 and removed in v26

NicolasLiampotis replied:
We can remove these lines since the proxy option was deprecated in Keycloak v24 and removed in v26

NicolasLiampotis commented on roles/keycloak/templates/26.4/keycloak.conf.j2:
```diff
@@ -0,0 +1,41 @@
+# The proxy address forwarding mode if the server is behind a reverse proxy.
+#proxy=reencrypt
+#proxy=edge
+proxy-headers=xforwarded
```
Replace with:
```
{% if keycloak_proxy_headers %}
# Configure how Keycloak interprets reverse proxy headers
proxy-headers={{ keycloak_proxy_headers }}
{% endif %}
```

Also in `defaults/main.yml` add:
```
# How Keycloak interprets reverse proxy headers.
# Allowed values: "xforwarded" or "forwarded".
# Leave commented when no proxy (or TLS passthrough).
#keycloak_proxy_headers: forwarded
```
*Important*: For our existing v26 deployments we need to explicitly define:
`keycloak_proxy_headers: xforwarded` in `groups_vars/keycloak/main.yml`

ademone replied:
added

NicolasLiampotis commented on roles/keycloak/templates/26.4/keycloak.conf.j2:
```diff
@@ -0,0 +1,41 @@
+
+
+# cache=ispn
+# cache-config-file=cache-ispn-jdbc-ping.xml
```
If we comment out `cache-config-file=cache-ispn-jdbc-ping.xml` then keycloak will ignore the templated cache-ispn-jdbc-ping.xml. I think this. I'd suggest uncommenting both cache=ispn and cache-config-file=cache-ispn-jdbc-ping.xml as with previous versions of the template - unless it's necessary to change the cache configuration

NicolasLiampotis replied:
I've verified that `cache-config-file=cache-ispn-jdbc-ping.xml` is required for Keycloak to use the custom cache config. Here is the full config:
```
cache=ispn
# Allows to apply custom cache configuration (e.g. cache-ispn-jdbc-ping.xml.j2)
# without triggering the deprecation warning about modifying defaults.
# Required when using a custom Infinispan XML file in production.
cache-config-mutate=true
cache-config-file=cache-ispn-jdbc-ping.xml
```

NicolasLiampotis commented on roles/keycloak/templates/26.4/keycloak.conf.j2:
```diff
@@ -0,0 +1,41 @@
+# ------------------------------------------------------------------
+# 1. ALLOW PLAIN HTTP (NGINX does TLS)
+# ------------------------------------------------------------------
+http-enabled=true
```
I'd move this directly after the `http-certificaton-*` options:
```
# Enable HTTP for reverse proxy setups where TLS terminates at the proxy
http-enabled=true

ademone replied:
moved

NicolasLiampotis commented on roles/keycloak/templates/26.4/cache-ispn-jdbc-ping.xml.j2:
This file is ignored with the current keycloak.confg.j2 template

Commit:
Add requested changes for the PR

ademone reviewed (commented):
Added a new commit

Commit:
Move http after htt certs

NicolasLiampotis committed:
Fix Keycloak version parsing with optional patch [RCIAM-369]

NicolasLiampotis reviewed (changes requested):
The Keycloak version parsing has been updated to allow for `<MAJOR>.<MINOR>` format (see https://github.com/rciam/rciam-deploy/commit/e78127c930f352da8e944befb0a8a8821d239630). Templates for 26.4.2 are no longer needed.

Commit:
Update directory to keep version template

Commit:
Fix following space after xforwarded

The previous syntax version was adding a trailing space resulting a broken xforwarded conf

Commit:
Revisit proxy headers configuration, also update README and requirements

NicolasLiampotis reviewed (approved):
LGTM
